pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -1136,9 +971,9 @@ func (p *PriorityQueue) Delete(pod *v1.Pod) {
 	defer p.lock.Unlock()
 	p.DeleteNominatedPodIfExists(pod)
 	pInfo := newQueuedPodInfoForLookup(pod)
-	p.activeQLock.Lock()
-	defer p.activeQLock.Unlock()
-	if err := p.activeQ.Delete(pInfo); err != nil {
+	p.activeQ.getLock().Lock()
+	defer p.activeQ.getLock().Unlock()
+	if err := p.activeQ.unlocked().Delete(pInfo); err != nil {
```

Comment:
Similar point, why do we have `Delete` under `unlocked()`? 
Why do we have to protect L974-L982 with activeQ.lock, in addition to p.lock?